fix: don't duplicate path in URL to post on subdirectory installs.

askvortsov1 opened the pull request.

luceos commented:
Feels like this issue solves **one** instance of a problem, a problem that can cascade into other areas including 3rd party extensions. What is exactly the cause? Is baseUrl sometimes the domain + path and sometimes just the path?

dsevillamartin commented:
@luceos `baseUrl` has the URL in `config.php`, so it will always have the subfolder if there is one. `app.route` will return an URL with the subfolder as well because it is needed for navigation to the correct page (`/subfolder/d/...`).

davwheat commented on framework/core/js/src/forum/components/PostMeta.js:
```diff
@@ -55,6 +55,8 @@ export default class PostMeta extends Component {
-    return app.forum.attribute('baseUrl') + app.route.post(post);
+    // URL.hostname transformation is necessary to avoid duplicating
+    // path in subdirectory installs.
+    return URL(app.forum.attribute('baseUrl')).hostname + app.route.post(post);
```
This will not keep URLs which contain ports, nor the protocol. We should use `.origin` instead.


```suggestion
    return new URL(app.forum.attribute('baseUrl')).origin + app.route.post(post);
```

askvortsov1 committed:
fix: don't duplicate path in URL to post on subdirectory installs.

`baseUrl` includes the path portion of the url, as does `basePath`, which is included in `app.route.x` function outputs. The `baseOrigin` (with no path component at all) should be used instead.
